src/json-api-serializer.js

```diff
@@ -234,17 +234,27 @@ function hasManyLink(key, type, record, attr) {
   return { linkage: linkages };
 }
 
+function getIdObject(linkage) {
+  if (linkage.id && linkage.type) {
+    return {
+      id: linkage.id,
+      type: Ember.String.camelize(linkage.type.singularize())
```

Comment:
I don't think this will work when Ember's prototype extensions are not added. The safe way to do this is to use `Ember.String.singularize`, I believe.

`String.prototype` only has `singularize` added when [prototype extensions are on](https://github.com/stefanpenner/ember-inflector/blob/e19798c3759f2e46bc2b9ce7de36b85397b5d824/addon/lib/ext/string.js#L24), but `Ember.String.singularize` is available when using ember-data all time time (it is set here, in the ember-inflector addon that ember-data uses: https://github.com/stefanpenner/ember-inflector/blob/e19798c3759f2e46bc2b9ce7de36b85397b5d824/addon/index.js#L16).